pep-0601.txt

```diff
@@ -1,7 +1,7 @@
 PEP: 601
 Title: Forbid return/break/continue breaking out of finally
 Author: Damien George, Batuhan Taskaya
-Sponsor: Nick Coghlan
+Sponsor: Nick Coghlan <ncoghlan@gmail.com>
```

Comment:
I asked Nick and he said it's fine too but the only reason he didn't add it is because he was just the sponsor here.

And checking [PEP 1](https://peps.python.org/pep-0001/#pep-header-preamble), only the author header mentions email. Sponsor and PEP-Delegate are listed as name only:

```
  Author: <list of authors' real names and optionally, email addrs>
* Sponsor: <real name of sponsor>
* PEP-Delegate: <PEP delegate's real name>
```

However, there are some existing PEPs with emails in Sponsor and *-Delegate...

- Sponsor with email: 31
- Sponsor without email: 12
- Delegate with email: 65
- Delegate without email: 41

Should PEP 1 be updated to allow optional emails in these headers? Or the emails removed? Something else?

Reply:
This is one of the (multiple) inconsistencies between PEP 1 and PEP 12 (and the template file), particularly when it comes to headers. [PEP 12 says](https://peps.python.org/pep-0012/#how-to-use-this-template)

```email
Author: [Full Name <email at example.com>]
Sponsor: *[Full Name <email at example.com>]
PEP-Delegate:
```

Since PEP 12 is more up to date and more precisely specified, I generally give it precedence, and I don't think there's a harm in allowing but not requiring an email to be specified, which is reflected in the updated checks in this PR.

> Should PEP 1 be updated to allow optional emails in these headers? Or the emails removed? Something else?

What I was going to do, as part of the larger updates planned, was to consolidate the repetitive and conflicting listings and descriptions of the headers and sections into PEP 12 and sync the template with it, focusing that PEP more tightly on the technical implementation details and author-focused guidance, while PEP 1 remains dedicated to describing the PEP _process_ and governance as a whole.

> Sponsor without email: 12

Huh; as a test I initially ran the check with the email required (which is how I found these), and it only identified four sponsors without emails, all of whom had them in multiple other PEPs (so I opportunistically updated them) as well as a few other issues. So it seems either your regex has false positives, or more concerning my regex (the one used here, just without the email optional) had false negatives, which is worrisome...

Reply:
Updating PEPs 1 and 12: sounds reasonable.

Sponsor without email count: Yeah, quite likely some false positives, just did some quick greps for a rough comparison, so I wouldn't worry about it too much. Main point is there's quite a few _with_ email.

Reply:
To note, I initially planned to de-conflict them with this PR, or in a predecessor one, but decided it would be best to separate that off and probably do it as part of the planned consolidation, rather than duplicating effort three places.